bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -599,37 +599,43 @@ public static boolean checkIsType(Type sourceType, Type targetType, List<TypePai
             return isUnionTypeMatch((BUnionType) sourceType, targetType, unresolvedTypes);
         }
 
+        if (sourceTypeTag == TypeTags.FINITE_TYPE_TAG &&
+                (targetTypeTag <= TypeTags.NULL_TAG || targetTypeTag == TypeTags.XML_TEXT_TAG)) {
```

Comment:
should we check only XML_TEXT_TAG here? Should we use TypeTags.isXMLTypeTag?

Reply:
Do we have tests for xml?

Reply:
`xml:Text` is an inherently immutable type and other `xml` subtypes are selectively immutable. IINM, the finite type is considered as inherently immutable type and the value spaces of finite types should contain inherently immutable types only. So, we do not need to have other `xml` subtypes under this check.